pkg/analysis_server/test/analysis/notification_closingLabels_test.dart

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:analysis_server/protocol/protocol.dart';
+import 'package:analysis_server/protocol/protocol_constants.dart';
+import 'package:analysis_server/protocol/protocol_generated.dart';
+import 'package:test/test.dart';
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../analysis_abstract.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(_AnalysisNotificationClosingLabelsTest);
+  });
+}
+
+@reflectiveTest
+class _AnalysisNotificationClosingLabelsTest extends AbstractAnalysisTest {
```

Comment:
You should include some tests in which some of the type names are prefixed (which will require, iirc, using prefixed named constructors like `new p.C.c(...)`).

That said, I would break these tests up a bit. Having a couple of tests that create a server, subscribe for and listen for notifications, and compare the result (what you have here) is good. But I'd move most of the tests into a set of unit tests for `DartUnitClosingLabelsComputer` because there's less overhead, so the tests will run faster. Not a problem if you don't agree or don't want to do that.